Functions/ImageProcessing/xASL_im_ASLSubtractionAveraging.m

```diff
@@ -92,6 +92,33 @@
 	error('Empty Path_ASL4D, PWI4D, and PWI3D');
 elseif (numel(path_ASL4D)~=1 && ~isnumeric(path_ASL4D) && numel(path_ASL4D)<1000)
     % this is a path, not an image or it is empty
+    
+    if xASL_exist(path_ASL4D, 'file')
+        % if this is a path, we will try to get the JSON sidecar
+        [pDir, pFile] = xASL_fileparts(path_ASL4D);
+        path_ASL4Djson = fullfile(pDir, [pFile '.json']);
+
+        if xASL_exist(path_ASL4Djson, 'file')
+            % we try to load this json and extract quantification parameters
+            json = xASL_io_ReadJson(path_ASL4Djson);
+            % if it has the two required quantification fields PLD & LD, we will use this
+            jsonFields = fields(json);
```

Comment:
You can remove these lines and looking for fields. We have converted to Legacy so we know that PLD has to be either at
JSON.Q.Initial_PLD or it isn't there.

So we simply remove lines 105-119 and have instead this. With the changes above, we'll already be in correct Legacy notation and format...
if isfield(json, 'Q') && isfield(json.Q, 'LabelingDuration')
   x.Q.Initial_PLD = json.Q.Initial_PLD;
   end 
   
   Same for EchoTime and LabelingDuration

Reply:
No we still have to check. The code actually becomes a bit longer, because we have to check for the existence of the field Q. Otherwise, the code is identical.

Reply:
I mean a different thing. It makes no sense to look for general PLD field as it **has** to be in Q.PLD and nowhere else.
Also - you need to change the notation as mentioned above. The parms2BIDS conversion changes the notation to Legacy and you are looking for BIDS fields.

Reply:
Fixed